ipatests/test_integration/test_installation.py

```diff
@@ -404,6 +404,46 @@ def test_install_dns(self):
             extra_args=['--dnssec-master', '--no-dnssec-validation']
         )
 
+    def test_ipactl_restart_pki_tomcat(self):
+        """ Test if ipactl restart restarts the pki-tomcatd
+
+        Wrong logic was triggering the start instead of restart
+        for pki-tomcatd. This test validates that restart
+        called on pki-tomcat properly.
+
+        related ticket : https://pagure.io/freeipa/issue/7927
+        """
+        splitted = []
+        # get process id of pki-tomcatd
+        cmd = self.master.run_command(['ps', '-ef'])
+        for line in cmd.stdout_text.split('\n'):
+            if "pki-tomcatd" in line:
+                splitted = line.split()
+        pki_pid = splitted[1]
```

Comment:
Why not use systemctl status pki-tomcatd@pki-tomcat and look for "Main PID:"?

pidof java would work too but could be fragile I suppose if we ended up with another java program (perhaps not likely)

Reply:
I tried with systemctl but was getting:
`SyntaxError: Non-ASCII character '\xe2' in file /tmp/temp.py on line 1, but no encoding declared; see http://www.python.org/peps/pep-0263.html for details`

`systemctl status` output have some ASCII characters, and need to use an encoding. I am not sure if using encoding will break other things. So I continued with `ps -ef`